### PR TITLE
Updated and fixed React to current standard

### PR DESCRIPTION
**Pull request by DanielAmorimAraujo**
- removed deprecated methods and replaced them with current methods
- updated to the standard syntax
- fixed all broken links

### CODE REVIEW

**Comment by kapnobatai137** [comment]
Thank you for your contribution @DanielAmorimAraujo.

**Comment by DanielAmorimAraujo** [comment]
Thanks, I just use your app a lot, but noticed that a lot of the information for React was outdated, so I decided to try and fix it. If anything's incorrect or I need to do something, just let me know!

File: javascript/react/props-iii/shallow-compare-in-react.md [hunk 1]
```diff
@@ -61,7 +61,7 @@ The **NotPure** component would render every time **this.props.className** updat
 
 Note that `PureComponent` only shallowly compares the objects, meaning it the **prop**s or **state** contains complex data structures, it may produce false-negatives for deeper differences. Only use `PureComponent` if the component has simple **prop**s and **state**.
```
Comment (DanielAmorimAraujo) [inline javascript/react/props-iii/shallow-compare-in-react.md]: Just noticed a mistake. It should say "meaning _if_ the **prop**s" not "meaning _it_ the **prop**s"

Comment (kapnobatai137) [reply javascript/react/props-iii/shallow-compare-in-react.md]: nice spot 👍

**Comment by enki-hq** [comment]
### Insight previews:
- [ ] [enkidevs/curriculum/.archived/javascript/react/good-to-know/keyed-fragments-in-react.md](https://app.enkipro.com/#/public/preview-insight/enkidevs/curriculum/.archived/javascript/react/good-to-know/keyed-fragments-in-react.md?ref=332e8ed58fd8bc98a59c0faf3cd00695afbebf94)
- [ ] [enkidevs/curriculum/.archived/javascript/react/props-ii/props-in-getinitialstate-is-an-anti-pattern.md](https://app.enkipro.com/#/public/preview-insight/enkidevs/curriculum/.archived/javascript/react/props-ii/props-in-getinitialstate-is-an-anti-pattern.md?ref=332e8ed58fd8bc98a59c0faf3cd00695afbebf94)
- [ ] [enkidevs/curriculum/.archived/javascript/react/props-iii/componentwillreceiveprops-not-triggered-after-mounting.md](https://app.enkipro.com/#/public/preview-insight/enkidevs/curriculum/.archived/javascript/react/props-iii/componentwillreceiveprops-not-triggered-after-mounting.md?ref=332e8ed58fd8bc98a59c0faf3cd00695afbebf94)
- [ ] [enkidevs/curriculum/javascript/react/basics-i/component-lifecycle-methods.md](https://app.enkipro.com/#/public/preview-insight/enkidevs/curriculum/javascript/react/basics-i/component-lifecycle-methods.md?ref=332e8ed58fd8bc98a59c0faf3cd00695afbebf94)
- [ ] [enkidevs/curriculum/javascript/react/basics-i/the-component-lifecycle.md](https://app.enkipro.com/#/public/preview-insight/enkidevs/curriculum/javascript/react/basics-i/the-component-lifecycle.md?ref=332e8ed58fd8bc98a59c0faf3cd00695afbebf94)
- [ ] [enkidevs/curriculum/javascript/react/basics-ii/conditional-rendering-part-2.md](https://app.enkipro.com/#/public/preview-insight/enkidevs/curriculum/javascript/react/basics-ii/conditional-rendering-part-2.md?ref=332e8ed58fd8bc98a59c0faf3cd00695afbebf94)
- [ ] [enkidevs/curriculum/javascript/react/basics-ii/jsx-spread-attributes.md](https://app.enkipro.com/#/public/preview-insight/enkidevs/curriculum/javascript/react/basics-ii/jsx-spread-attributes.md?ref=332e8ed58fd8bc98a59c0faf3cd00695afbebf94)
- [ ] [enkidevs/curriculum/javascript/react/dummy-workout/use-react-with-other-libraries.md](https://app.enkipro.com/#/public/preview-insight/enkidevs/curriculum/javascript/react/dummy-workout/use-react-with-other-libraries.md?ref=332e8ed58fd8bc98a59c0faf3cd00695afbebf94)
- [ ] [enkidevs/curriculum/javascript/react/fundamentals/the-react-component.md](https://app.enkipro.com/#/public/preview-insight/enkidevs/curriculum/javascript/react/fundamentals/the-react-component.md?ref=332e8ed58fd8bc98a59c0faf3cd00695afbebf94)
- [ ] [enkidevs/curriculum/javascript/react/good-to-know/fragments-in-react.md](https://app.enkipro.com/#/public/preview-insight/enkidevs/curriculum/javascript/react/good-to-know/fragments-in-react.md?ref=332e8ed58fd8bc98a59c0faf3cd00695afbebf94)
- [ ] [enkidevs/curriculum/javascript/react/good-to-know/immutability-helpers-in-react.md](https://app.enkipro.com/#/public/preview-insight/enkidevs/curriculum/javascript/react/good-to-know/immutability-helpers-in-react.md?ref=332e8ed58fd8bc98a59c0faf3cd00695afbebf94)
- [ ] [enkidevs/curriculum/javascript/react/good-to-know/load-initial-data-via-ajax.md](https://app.enkipro.com/#/public/preview-insight/enkidevs/curriculum/javascript/react/good-to-know/load-initial-data-via-ajax.md?ref=332e8ed58fd8bc98a59c0faf3cd00695afbebf94)
- [ ] [enkidevs/curriculum/javascript/react/good-to-know/reactdom-render-ref.md](https://app.enkipro.com/#/public/preview-insight/enkidevs/curriculum/javascript/react/good-to-know/reactdom-render-ref.md?ref=332e8ed58fd8bc98a59c0faf3cd00695afbebf94)
- [ ] [enkidevs/curriculum/javascript/react/gotchas/dangerously-set-innerhtml.md](https://app.enkipro.com/#/public/preview-insight/enkidevs/curriculum/javascript/react/gotchas/dangerously-set-innerhtml.md?ref=332e8ed58fd8bc98a59c0faf3cd00695afbebf94)
- [ ] [enkidevs/curriculum/javascript/react/gotchas/if-else-statements-in-jsx-and-react.md](https://app.enkipro.com/#/public/preview-insight/enkidevs/curriculum/javascript/react/gotchas/if-else-statements-in-jsx-and-react.md?ref=332e8ed58fd8bc98a59c0faf3cd00695afbebf94)
- [ ] [enkidevs/curriculum/javascript/react/gotchas/react-in-line-styles.md](https://app.enkipro.com/#/public/preview-insight/enkidevs/curriculum/javascript/react/gotchas/react-in-line-styles.md?ref=332e8ed58fd8bc98a59c0faf3cd00695afbebf94)
- [ ] [enkidevs/curriculum/javascript/react/gotchas/using-false-in-jsx.md](https://app.enkipro.com/#/public/preview-insight/enkidevs/curriculum/javascript/react/gotchas/using-false-in-jsx.md?ref=332e8ed58fd8bc98a59c0faf3cd00695afbebf94)
- [ ] [enkidevs/curriculum/javascript/react/intro-react/what-i-need-to-use-react.md](https://app.enkipro.com/#/public/preview-insight/enkidevs/curriculum/javascript/react/intro-react/what-i-need-to-use-react.md?ref=332e8ed58fd8bc98a59c0faf3cd00695afbebf94)
- [ ] [enkidevs/curriculum/javascript/react/intro-react/what-is-react.md](https://app.enkipro.com/#/public/preview-insight/enkidevs/curriculum/javascript/react/intro-react/what-is-react.md?ref=332e8ed58fd8bc98a59c0faf3cd00695afbebf94)
- [ ] [enkidevs/curriculum/javascript/react/intro-react/why-use-react.md](https://app.enkipro.com/#/public/preview-insight/enkidevs/curriculum/javascript/react/intro-react/why-use-react.md?ref=332e8ed58fd8bc98a59c0faf3cd00695afbebf94)
- [ ] [enkidevs/curriculum/javascript/react/props-i/enhanced-prop-validation.md](https://app.enkipro.com/#/public/preview-insight/enkidevs/curriculum/javascript/react/props-i/enhanced-prop-validation.md?ref=332e8ed58fd8bc98a59c0faf3cd00695afbebf94)
- [ ] [enkidevs/curriculum/javascript/react/props-i/prop-validation.md](https://app.enkipro.com/#/public/preview-insight/enkidevs/curriculum/javascript/react/props-i/prop-validation.md?ref=332e8ed58fd8bc98a59c0faf3cd00695afbebf94)
- [ ] [enkidevs/curriculum/javascript/react/props-ii/children-in-react.md](https://app.enkipro.com/#/public/preview-insight/enkidevs/curriculum/javascript/react/props-ii/children-in-react.md?ref=332e8ed58fd8bc98a59c0faf3cd00695afbebf94)
- [ ] [enkidevs/curriculum/javascript/react/props-ii/dealing-with-this-props-children.md](https://app.enkipro.com/#/public/preview-insight/enkidevs/curriculum/javascript/react/props-ii/dealing-with-this-props-children.md?ref=332e8ed58fd8bc98a59c0faf3cd00695afbebf94)
- [ ] [enkidevs/curriculum/javascript/react/props-ii/props-in-constructor-state-is-an-anti-pattern.md](https://app.enkipro.com/#/public/preview-insight/enkidevs/curriculum/javascript/react/props-ii/props-in-constructor-state-is-an-anti-pattern.md?ref=332e8ed58fd8bc98a59c0faf3cd00695afbebf94)
- [ ] [enkidevs/curriculum/javascript/react/props-ii/specify-a-single-child.md](https://app.enkipro.com/#/public/preview-insight/enkidevs/curriculum/javascript/react/props-ii/specify-a-single-child.md?ref=332e8ed58fd8bc98a59c0faf3cd00695afbebf94)
- [ ] [enkidevs/curriculum/javascript/react/props-ii/type-of-the-children-props.md](https://app.enkipro.com/#/public/preview-insight/enkidevs/curriculum/javascript/react/props-ii/type-of-the-children-props.md?ref=332e8ed58fd8bc98a59c0faf3cd00695afbebf94)
- [ ] [enkidevs/curriculum/javascript/react/props-iii/clone-elements-in-react.md](https://app.enkipro.com/#/public/preview-insight/enkidevs/curriculum/javascript/react/props-iii/clone-elements-in-react.md?ref=332e8ed58fd8bc98a59c0faf3cd00695afbebf94)
- [ ] [enkidevs/curriculum/javascript/react/props-iii/shallow-compare-in-react.md](https://app.enkipro.com/#/public/preview-insight/enkidevs/curriculum/javascript/react/props-iii/shallow-compare-in-react.md?ref=332e8ed58fd8bc98a59c0faf3cd00695afbebf94)
- [ ] [enkidevs/curriculum/javascript/react/props-iii/shortcut-for-transferring-props.md](https://app.enkipro.com/#/public/preview-insight/enkidevs/curriculum/javascript/react/props-iii/shortcut-for-transferring-props.md?ref=332e8ed58fd8bc98a59c0faf3cd00695afbebf94)
- [ ] [enkidevs/curriculum/javascript/react/props-iii/use-proptypes-on-stateless-components.md](https://app.enkipro.com/#/public/preview-insight/enkidevs/curriculum/javascript/react/props-iii/use-proptypes-on-stateless-components.md?ref=332e8ed58fd8bc98a59c0faf3cd00695afbebf94)

**Comment by DanielAmorimAraujo** [comment]
Hey @nem035, I agree with all your suggestions. The language sounds cleaner and more concise. And thanks for fixing some of the indentation, as I wasn't 100% sure how things were to be formatted.

File: javascript/react/intro-react/what-i-need-to-use-react.md [hunk 2]
```diff
@@ -91,7 +91,7 @@ Which tool will automatically set-up the development environment for a simple `R
 A `JavaScript` **bundler** is a tool that puts your code and all its dependencies together in one `JS` file. There are many of them out there these days, being the most popular ones `browserify` and `webpack`.
 
 [2:Babel]
-`Babel` is a compiler that transforms newer `JS` sytnax (such as `ES6`) into standard `JS` that can run on the browser(or server).
+`Babel` is a compiler that transforms newer `JS` syntax (such as `ES6`) into standard `JS` that can run on the browser(or server).
```
[the remark this answers is not in the document]
Comment (DanielAmorimAraujo) [reply javascript/react/intro-react/what-i-need-to-use-react.md]: I didn't even write this section (its part of the previous code) but I do agree

**Pull request merged**